fdk_extension/webhook.py

```diff
@@ -0,0 +1,354 @@
+"""Webhook utility."""
+import hashlib
+import hmac
+import re
+
+import ujson
+
+
+from .constants import ASSOCIATION_CRITERIA, TEST_WEBHOOK_EVENT_NAME
+from .exceptions import FdkInvalidHMacError
+from .exceptions import FdkInvalidWebhookConfig
+from .exceptions import FdkWebhookHandlerNotFound
+from .exceptions import FdkWebhookProcessError
+from .exceptions import FdkWebhookRegistrationError
+from .utilities.logger import get_logger
+
+from fdk_client.common.aiohttp_helper import AiohttpHelper
+from fdk_client.common.utils import get_headers_with_signature
+from fdk_client.platform.PlatformClient import PlatformClient
+
+from sanic.request import Request
+
+
+logger = get_logger()
+
+event_config = {}
+
+class WebhookRegistry:
+    def __init__(self):
+        self._handler_map = None
+        self._config : dict = None
+        self._fdk_config : dict = None
+
+    async def initialize(self, config: dict, fdk_config: dict):
+        email_regex_match = r"^\S+@\S+\.\S+$"
+        if not config.get("notification_email") or not re.findall(email_regex_match, config["notification_email"]):
+            raise FdkInvalidWebhookConfig("Invalid or missing notification_email")
+
+        if not config.get("api_path") or config["api_path"][0] != "/":
+            raise FdkInvalidWebhookConfig("Invalid or missing api_path")
+
+        if not config.get("event_map"):
+            raise FdkInvalidWebhookConfig("Invalid or missing event_map")
+
+        config["subscribe_on_install"] = True if not config.get("subscribe_on_install") else config["subscribe_on_install"]
```

Comment:
use default value of getter here. We don't have getter functions on JS hence added ternary operator.